### ISSUE TITLE
Should have unique constraint for Field name + table_id + parent_id and for Table name + schema + db_id

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#669]
We don't want to have multiple definitions for the same `Field`. We should add a DB constraint to make sure that can't happen

**Comment by salsakran** [comment]
@camsaul is this still a problem?

**Comment by camsaul** [comment]
As with #668 I haven't seen this cause any real-world issues in a long time, but there's still no DB-level constraint, and it would be nice to have one.

**Comment by camsaul** [comment]
So this DB constraint should be easy to add but to be extra-sure we don't break anyone's instances we need to write a query that clears any duplicate entries. This works for Postgres:

```sql
DELETE
FROM metabase_field f 
WHERE (
    SELECT COUNT(*)
    FROM metabase_field f2
	WHERE f.table_id = f2.table_id
      AND ((CASE WHEN f.parent_id IS NULL THEN 0 ELSE f.parent_id END) =
           (CASE WHEN f2.parent_id IS NULL THEN 0 ELSE f2.parent_id END))
      AND f.name = f2.name
    )::integer > 1
```

I'll have to figure out MySQL / H2.

**Comment by camsaul** [comment]
Or something like this

```sql
DELETE
FROM metabase_field 
WHERE id IN (
    SELECT id 
    FROM metabase_field f1
	INNER JOIN (
		SELECT table_id, (CASE WHEN parent_id IS NULL THEN 0 ELSE parent_id END), name
        FROM metabase_field
        GROUP BY table_id, parent_id, name
        HAVING count(*) > 1
    ) f2
    ON f1.table_id = f2.table_id
    AND (CASE WHEN f1.parent_id IS NULL THEN 0 ELSE f1.parent_id END) = f2.parent_id
    AND f1.name = f2.name
)
```

**Comment by camsaul** [comment]
Actually this seems to be the simplest way to do things. This should work on all 3 (other than capitalization):

```sql
DELETE
FROM metabase_field
WHERE (table_id, (CASE WHEN parent_id IS NULL THEN 0 ELSE parent_id END), name) IN (
    SELECT table_id, (CASE WHEN parent_id IS NULL THEN 0 ELSE parent_id END) AS parent_id, name
    FROM metabase_field
    GROUP BY table_id, parent_id, name
    HAVING count(*) > 1
)
```

**Comment by camsaul** [comment]
We should consider doing the same thing for Table + schema + db_id

**Comment by camsaul** [comment]
Implemented by #9220
